app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -127,39 +117,40 @@ public void setView(NewsFeedContract.View view) {
     public void invertFavoriteState() {
         showOnlyFavorite = !showOnlyFavorite;
         view.setFavoriteIcon(showOnlyFavorite);
-        loadNewsFromDB();
+        getNewsFromStorage();
         updateNewsListUI();
     }
 
     @Override
     public void updateNewsFromApi() {
-        loadNewsFromApi();
+        updateActualNews();
     }
 
-    private List<NewsItem> getNewsFromDB() {
-        return newsStorage.getFiltered(showOnlyFavorite, category);
+    private void getNewsFromStorage() {
+        StorageGetUseCase getUseCase = new StorageGetUseCase(newsStorage);
+        news = getUseCase.run(null, new Filter(category, showOnlyFavorite));
     }
 
-    private void requestNewsFromApi() {
-        repository.getNewsFiltered(category, new RemoteNewsRepository.RequestListener() {
-            @Override
-            public void onRequestSuccess(Response<NewsApiResponse> response) {
-                view.hideProgressBar();
-                if (!response.isSuccessful()) {
-                    Log.d(TAG, "onResponse " + response.code());
-                    return;
-                }
-                List<NewsItem> news = mapper.map(response, category);
-                newsStorage.insertUnique(news);
-                loadNewsFromDB();
-                updateNewsListUI();
-            }
+    private void updateNewsStorage() {
+        StorageUpdateUseCase updateUseCase = new StorageUpdateUseCase(newsStorage, repository, mapper );
+        updateUseCase.run(null, new Filter(category, showOnlyFavorite));
+    }
+    private void updateItemsStorage(List<NewsItem> updateList){
+        StorageUpdateItemUseCase updateItemUseCase = new StorageUpdateItemUseCase(newsStorage);
```

Comment:
The same thing about naming here. BTW, all your use cases should be injected but not created here.